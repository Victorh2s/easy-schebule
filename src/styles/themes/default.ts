export const defaultTheme = {
  Light: {
    button: '#5300FF',
    success: '#2AAA50',
    error: '#D42029',
    gray: '#A4A69F',
    'font-color': '#FFFFFF',
    'logo-color': '#05543A',
    'pendente-color': '#F73A4E',
    'fazendo-color': '#FF911C',
    'feito-color': '#2AE850',
    'background-color': '#0f0f0f',
    'link-color': '#15A89F',
    'background-form': '#171717',
    'border-input': '#D2D2D4',
  },
  Dark: {
    button: '#5300FF',
    success: '#2AAA50',
    error: '#D42029',
    gray: '#A4A69F',
    'font-color': '#0f0f0f',
    'logo-color': '#05543A',
    'pendente-color': '#F73A4E',
    'fazendo-color': '#FF911C',
    'feito-color': '#2AE850',
    'background-color': '#FFFFFF',
    'link-color': '#15A89F',
    'background-form': '#EDEDEF',
    'border-input': '#D2D2D4',
  },
} as const;
