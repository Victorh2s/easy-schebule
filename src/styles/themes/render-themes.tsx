import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from './default';
import { BrowserRouter } from 'react-router-dom';

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <BrowserRouter>
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>{' '}
    </BrowserRouter>
  );
};
