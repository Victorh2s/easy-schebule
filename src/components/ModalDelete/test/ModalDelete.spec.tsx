import { ModalDeleteTask } from '../index';
import { renderTheme } from '../../../styles/themes/render-themes';

describe('ModalDeleteTask', () => {
  test('deletes task and shows success message', async () => {
    const { getByText } = renderTheme(<ModalDeleteTask />);

    expect(
      getByText('Tem certeza que deseja deletar essa tarefa?')
    ).toBeInTheDocument();
    expect(getByText('Deletar')).toBeInTheDocument();
    expect(getByText('Fechar')).toBeInTheDocument();
  });

  test('should match to snapshot', () => {
    const { container } = renderTheme(<ModalDeleteTask />);

    expect(container).toMatchSnapshot();
  });
});
