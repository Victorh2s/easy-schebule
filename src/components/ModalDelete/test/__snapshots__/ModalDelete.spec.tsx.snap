// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModalDeleteTask > should match to snapshot 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-animation: jgQpwH 0.2s ease-in-out;
  animation: jgQpwH 0.2s ease-in-out;
}

.c1 {
  background-color: #171717;
  width: 50%;
  height: 35%;
  padding: 4rem;
  border-radius: 5px;
  -webkit-animation: jgQpwH 0.2s ease-in-out;
  animation: jgQpwH 0.2s ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  overflow-x: hidden;
}

.c1 h2 {
  width: 100%;
  text-align: center;
}

.c1 div button:first-child:hover {
  background: #2AAA50;
}

.c1 div button {
  border-radius: 0.2rem;
  border: none;
  top: 0;
  right: 0;
  padding: 0.5rem;
  cursor: pointer;
  background: #5300FF;
  color: #FFFFFF;
  font-weight: bold;
  margin-top: 1rem;
  margin-right: 1rem;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c1 div button:disabled {
  background: #A4A69F !important;
  cursor: not-allowed;
}

.c1 div button:hover {
  background: #F73A4E;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h2>
        Tem certeza que deseja deletar essa tarefa?
      </h2>
      <div>
        <button>
          Deletar
        </button>
        <button>
          Fechar
        </button>
      </div>
    </div>
  </div>
   
</div>
`;
